docs/docs/integrations/providers/mlflow_tracking.ipynb

```diff
@@ -47,142 +57,498 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "%pip install --upgrade --quiet  azureml-mlflow\n",
-    "%pip install --upgrade --quiet  pandas\n",
-    "%pip install --upgrade --quiet  textstat\n",
-    "%pip install --upgrade --quiet  spacy\n",
-    "%pip install --upgrade --quiet  langchain-openai\n",
-    "%pip install --upgrade --quiet  google-search-results\n",
-    "!python -m spacy download en_core_web_sm"
+    "%pip install langchain-openai -qU"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
-   "id": "bf8e1f5c",
+   "execution_count": 6,
+   "id": "7e87b21d",
    "metadata": {},
    "outputs": [],
    "source": [
     "import os\n",
     "\n",
+    "# Set MLflow tracking URI if you have MLflow Tracking Server running\n",
     "os.environ[\"MLFLOW_TRACKING_URI\"] = \"\"\n",
-    "os.environ[\"OPENAI_API_KEY\"] = \"\"\n",
-    "os.environ[\"SERPAPI_API_KEY\"] = \"\""
+    "os.environ[\"OPENAI_API_KEY\"] = \"\""
```

Comment:
Can we validate that `OPENAI_API_KEY` is set if this notebook fails without `OPENAI_API_KEY`?

Reply:
@haru If it's not set, `ChatOpenAI` instantiation fails with the following error, which looks clear enough. Do you think we should add another validation here?

```
Did not find openai_api_key, please add an environment variable `OPENAI_API_KEY` which contains it, or pass `openai_api_key` as a named parameter. (type=value_error)
```

Reply:
We have that assertion validation all over our demo notebooks in MLflow docs. I think it's fine to allow the underlying library to raise.